_posts/2019-02-14-supervisor-strategies.md

```diff
@@ -0,0 +1,358 @@
+---
+author: Bobby Grayson
+author_link: https://github.com/notactuallypagemcconnell
+date: 2019-02-14
+layout: post
+title:  Elixir Supervisor Strategies
+excerpt: >
+  Learn the ins and outs of Elixir's 3 supervisor strategies
+---
+
+One of the things that makes OTP and Elixir unique is the model of supervisor behaviour that applications can take with different processes they start.
+In this post we will examine each of the three available in Elixir by making a supervised app.
+
+To start, we make a supervised application:
+
+```
+mix new counter --sup
+cd counter
+```
+
+Now that we have an app, we are going to create 3 modules.
+They will all be GenServers that are started with the application who send themselves a message every second to increment their state by one.
+One will always work, one will fail every 6 messages, and one will fail every 20 messages.
+To start, it will have the default supervisory strategy of `one_for_one` in `application.ex`.
+This strategy says that if one process dies, its siblings should stay working unaffected.
+
+**Note: No matter your supervisory strategy, if the children in your app do not succeed on ```start_link``` and return an ```{:ok, pid}``` tuple, the application as a whole will not start and your supervisory strategy does not matter at all.**
+
+We will stick with that to start.
+
```

Comment:
Change one of the 'starts' to something else (e.g. `with that in the beginning.` or `Let's begin with`.